docs/REST API/rest-api.md

```diff
@@ -4551,6 +4572,40 @@ To wipe a macOS, iOS, iPadOS, or Windows host, the host must have MDM turned on.
 ```json
 {
   "activities": [
+    // uninstalled_software failed
+    {
+      "created_at": "2023-07-27T14:35:08Z",\
+      "actor_id": 1,
+      "actor_full_name": "Anna Chao",
+      "id": 1,
+      "actor_gravatar": "",
+      "actor_email": "",
+      "type": "uninstalled_software",
+      "details": {
+        "host_id": 1,
+        "host_display_name": "Marko’s MacBook Pro",
+        "software_title": "Adobe Acrobat.app",
+        "script_execution_id": "ecf22dba-07dc-40a9-b122-5480e948b756",
+        "status": "failed"
```

Comment:
@getvictor @noahtalerman this "failed" value throws a bit of a curveball – having the potential values here match those from software titles / [below](https://github.com/fleetdm/fleet/pull/21548/files#diff-7246bc304b15c8865ed8eaa205e9c244d0a0314e4bae60cf553dc06147c38b64R4688) would both make intuitive sense and help simplify/unify the frontend code:
`export const SOFTWARE_INSTALL_STATUSES = [
  "installed",
  "uninstalled",
  "pending_install",
  "failed_install",
  "pending_uninstall",
  "failed_uninstall",
] as const;`

Does that make sense from your points of view?

Reply:
Well, `uninstalled` is not a software install status. After software is uninstalled, the status goes to `null`, which means available for install.

Reply:
This type is used in multiple places both for API response typing and various display conversions, so its definition doesn't directly map to the responses. The purpose was to suggest `failed_uninstall` instead of `failed`, do you think that makes sense?

Reply:
I'm OK with `failed_uninstall`. But what do you want as the activity status when uninstall succeeded?

Reply:
`uninstalled`?

Reply:
which is what it currently is:
![Screenshot 2024-09-10 at 11 59 06 AM](https://github.com/user-attachments/assets/701670e7-000a-48ff-8d50-6403e05ab2a9)

Reply:
The array above, by the way, is being used by the frontend in a number of ways, including generating types for activities, not just to define the API response types

Reply:
Hey @jacobshandling and @getvictor heads up that I think we went w/ one `failed` to be consistent w/ configuration profile status.

Failed install and failed removal for profiles are both `failed`:
![Screenshot 2024-09-11 at 10 00 22 AM](https://github.com/user-attachments/assets/779eab97-42eb-41fb-bf29-1e827fbaf092)

Not sure if that still makes sense here but just wanted to call it out.

Reply:
@noahtalerman thanks for the added context. I'll look into making it work as specified with "failed" today